paymentsheet/src/main/java/com/stripe/android/lpmfoundations/paymentmethod/PaymentMethodMetadata.kt

```diff
@@ -69,7 +73,7 @@ internal data class PaymentMethodMetadata(
     val linkConfiguration: PaymentSheet.LinkConfiguration,
     val paymentMethodSaveConsentBehavior: PaymentMethodSaveConsentBehavior,
     val linkMode: LinkMode?,
-    val linkState: LinkState?,
+    val linkStateResult: LinkStateResult?,
```

Comment:
👀 Introduced `LinkStateResult`, which includes a disabled state with reasons. `linkState` is below as a derived property.

Note that `LinkStateResult` is a sealed interface with that the existing `LinkState` now implements. This may be a bit unusual, but it helps to avoid a ton of refactoring (a common theme in this PR).